### ISSUE TITLE
Simplify My Request Detail page code and update tests

### ISSUE THREAD

**Issue by pjmonks** [MauroDataMapper/mdm-explorer#258]
Issue created to split up #244 

After #202 was merged, a lot of tests needed to be added back in. Issue #244 has added a lot more tests, but the My Request Detail page has become more complex as a result of adding schema/class/element grouping to lists. To avoid creating a larger PR for #244, split this particular task up as standalone.

Look at:

1. Refactoring the code behind the My Request Detail page - can it be simplified at all?
2. Add back tests to cover the scenarios of this page.
3. Compare with the Template Detail page too - this will be similar but a cutdown version.

**Comment by pjmonks** [comment]
After hearing anecdotally how large requests can break the My Request Detail screen, I'm wondering if #257 is related to this issue regarding code complexity.

**Comment by CristianGimenez** [comment]
#257 is now solved, I think it has simplified a bit, but not massively, my-requests-detail still holds a lot of different functionalities... copy elements to new requests, delete elements... etc

**Comment by CristianGimenez** [comment]
There is too many people working on this page at the moment, so I'm moving to another task. These are the notes I've taken so far related to this component:


- 	 initialiseRequest and initialiseRequestQueries should be private. Also, both have subscriptions, one happens after the other, should we combine those into the same using map? Would that be better?
- 	selectOrUnselectAllDataElements should also be private
- 	Could we move the deletion to each of the elements, and just refresh? Would refreshing each component content update the my-request details view?
- 	 showCohortCreate, showCohortEdit and showDataEdit can be removed, they are not in use
- setRequest, firstly updates this.request (without checking for null) then, if its not null, it does things. Is that right? Should we update this.request with null?

**Comment by pjmonks** [comment]
@NigelPalmer @CristianGimenez I think this is now unblocked again, all current PRs for the My Request Details page should be merged now.

**Comment by CristianGimenez** [comment]
Apparently there is no consensus about refactoring the component to make it smaller. So, I've used this issue to include and fix the commented tests for the page. These are my notes in case we want to revisit this at any point in the future.

### GENERAL IMPROVEMENTS

- [ ] `submitRequest()` -> There is a filter, but the response there does not seem to be used, in `switchMap `we don't get any parameters.
- [ ] `removeSelected()` -> broadcast loading false should happen in a finalize statement rather than a subscribe? 
- [x] `showCohortCreate`, `showCohortEdit `and `showDataEdit ` can be removed, they are not in use
- [x] `initialiseRequest`, `initialiseRequestQueries`, `selectOrUnselectAllDataElements`, should be private
- [ ] `setRequest`, firstly updates `this.request` (without checking for null) then, if its not null, it does things. Is that right? Should we update `this.request` with null?


### CODE REFACTOR

- [ ] `initialiseRequest `and `initialiseRequestQueries `have subscriptions, one happens after the other, should we combine those into the same using map? Would that be better?
- [ ] could `copyRequest()` and `editRequest()` be handled within `mdm-data-request-row`, and then raise an update request element with the new data we need to refresh the view?
- [ ] Could we move the deletion to each of the elements, and just refresh? Would refreshing each component content update the `my-request-details` view?
